init/devices.cpp

```diff
@@ -61,6 +61,18 @@ namespace init {
 
 const int Aliases::ANY = -1;
 
+std::string Aliases::ToString() const {
+    std::stringstream fmt;
+
+    fmt << "AliasTo:" << alias_to_ << " "
+        << "Major:" << (major_ == Aliases::ANY ? "Any" : std::to_string(major_)) << " "
+        << "Minor:" << (minor_ == Aliases::ANY ? "Any" : std::to_string(minor_)) << " "
+        << "ProductId:" << productId_ << " "
+        << "VendorId:" << vendorId_;
```

Comment:
Maybe add `<< std:hex` before appending the product and vendor id as hex is used pretty much everywhere else.

Also, (nitpick) the vendor generally appears before the product as in `lsusb`. We have it swapped in the config file also but arguably too late to change.

Reply:
> How are "ordinary" device symlink removed and what makes our links special?

Symlinks such as things created in /dev/block/by-name, come from other code paths, and there is already code to erase them. Ours, created through 'alias' configuration entries, needs to have the associated state, in /dev/aliases, to be erased before the link itself is erased.

Reply:
> Maybe add `<< std:hex` before appending the product and vendor id as hex is used pretty much everywhere else.

Good idea. I'll add that.

> Also, (nitpick) the vendor generally appears before the product as in `lsusb`. We have it swapped in the config file also but arguably too late to change.

I never noticed this. It's unfortunately a bit late to change it now.